### PR TITLE
Compression using all the cpu's

### PR DESCRIPTION
**Pull request by vijaymadhavan**
multithreaded compression for GeoTiff by using all available CPU's and specify number of colors in the palette

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -146,6 +147,7 @@ Command Line Interface
       --no-fill-borders     Do not add borders to fill image.
       --zoom-offset N       Offset zoom level by N to fit unprojected images to
                             square maps. Defaults to 0.
+      --png8                specifies number of colors in palette. value range from 2 to 256.
```
Comment (pauricthelodger) [inline README.rst]: @mghughes do this feel like a coloring argument to you?

File: gdal2mbtiles/helpers.py [hunk 2]
```diff
@@ -32,7 +32,7 @@
 def image_mbtiles(inputfile, outputfile, metadata,
                   min_resolution=None, max_resolution=None, fill_borders=None,
                   zoom_offset=None, colors=None, renderer=None,
-                  preprocessor=None):
+                  preprocessor=None,pngdata={}):
```
Comment (pauricthelodger) [inline gdal2mbtiles/helpers.py]: @vijaymadhavan, could you replace dict objects in the function definitions which an `if pngdata is None: pngdata = {}` inside the function blocks? The reason for this is that sometimes declaring a list or dict object in the kwargs can cause the data to leak into other calls of that function.

**Comment by pauricthelodger** [comment]
Thanks for the contributions so far @vijaymadhavan! The comment above is the main thing I see but I was also wondering if you could put some spacing between the variables for pep8iness? =)

File: gdal2mbtiles/gdal.py [hunk 3]
```diff
@@ -921,7 +921,7 @@ def get_tempfile(self, **kwargs):
         tempfile.seek(0)
         return tempfile
 
-    def render(self, outputfile, cmd=GDALTRANSLATE, working_memory=512,
+    def render(self, outputfile, cmd=GDALTRANSLATE, working_memory=1024,
```
Comment (mghughes) [inline gdal2mbtiles/gdal.py]: What's the reason for this change in `working_memory`?

File: README.rst (hunk 1 above)
Comment (mghughes) [reply README.rst]: I think it's fine in the rendering options

Comment (mghughes) [inline README.rst]: @vijaymadhavan Could you specify what the default behaviour for this argument is here?

**Comment by mghughes** [comment]
@vijaymadhavan Thanks very much for the contribution! I noted a couple things in my review, and I also agree with @pauricthelodger that there are some spacing/style changes that should probably be addressed. Just so we can keep a consistent codebase :)

File: README.rst (hunk 1 above)
Comment (vijaymadhavan) [reply README.rst]: will modify the help message in a more meaningful way

Comment (vijaymadhavan) [reply README.rst]: sure. will do

File: gdal2mbtiles/gdal.py (hunk 3 above)
Comment (vijaymadhavan) [reply gdal2mbtiles/gdal.py]: @mghughes gdal_translate use all the available CPU for rending. so need more memory for processing

**Comment by vijaymadhavan** [comment]
@pauricthelodger @mghughes I have made changes as suggested. pls have a look

File: gdal2mbtiles/gdal.py [hunk 4]
```diff
@@ -936,6 +936,7 @@ def render(self, outputfile, cmd=GDALTRANSLATE, working_memory=512,
                     '-q',                   # Quiet - FIXME: Use logging
                     '-of', 'GTiff',         # Output to GeoTIFF
                     '-co', 'BIGTIFF=IF_SAFER',  # Use BigTIFF if >2GB
+                    '-co', 'NUM_THREADS=ALL_CPUS', # multithreaded compression for GeoTiff
```
Comment (damon-rand) [inline gdal2mbtiles/gdal.py]: Shouldn't the warp_cmd parameter here be something like "-mt" (for multithreading) instead of "-co"?

**Comment by damon-rand** [comment]
Thanks @vijaymadhavan for making the changes. @pauricthelodger, @mghughes can you guys take a quick look.

**Comment by pauricthelodger** [comment]
Thanks for those changes, @vijaymadhavan!

File: gdal2mbtiles/gdal.py (hunk 4 above)
Comment (damon-rand) [reply gdal2mbtiles/gdal.py]: As @mghughes helpfully pointed out... `-co` stands for "creation option" of which there can be multiple. Ignore my comment above.
